Pre 1.0.0

boygirl opened the pull request:
@ryan-roemer @tptee 

This PR converts Victory to a Lerna monorepo. Since I've added git history from `victory-core`, `victory-pie` and `victory-chart` the commits and changes in this branch are ~~a bit~~ totally overwhelming and unhelpful.

Relevant files:

-`CONTRIBUTING.md`
-`package.json`
-`package-scripts.js`
-`lerna,json`
-`/config`

TODOS:
- wire up chromatic CI
- determine what remaining changes I need to make to have lerna correctly publish (probably move publishr scripts to individual `package/victory-*/package.json`)
- `test-dev` sometimes errors with a timeout

ryan-roemer commented on README.md:
```diff
@@ -150,4 +78,9 @@ For example, when creating a `VictoryLine` component, you can pass a `groupCompo
+
+[React]: https://facebook.github.io/react/
+[prop-types]: https://github.com/reactjs/prop-types
+trav_img]: https://api.travis-ci.org/FormidableLabs/victory.svg
```
NIT: Missing starting `[`

ryan-roemer commented on .travis.yml:
```diff
@@ -1,17 +1,20 @@
 node_js:
-- '6'
 - '8'
 - '9'
```
Ideally do `6`, `8`, `10` as that's the currently supported nodes.

ryan-roemer replied:
Separate note -- out node code can and should be node6+ going forward as a future ticket (definitely not needed here ;) )

ryan-roemer commented on config/karma/karma.conf.dev.js:
```diff
@@ -0,0 +1,46 @@
+ */
+module.exports = function (config) {
+  config.set({
+    frameworks: ["mocha", "phantomjs-shim"],
```
Suggest switching to headless chrome instead. Phantomjs is on it's way out and headless chrome is so much more reliable. (May solve your test-dev timeouts). 

For an example, this pr switched Radium to headless chrome (and other unrelated stuff); https://github.com/FormidableLabs/radium/pull/962/files

ryan-roemer commented on config/webpack/demo/webpack.config.dev.js:
```diff
@@ -0,0 +1,62 @@
+module.exports = {
+
+  devServer: {
+    port: parseInt(process.env.npm_package_config_wds_port_dev || WDS_PORT),
```
Are we using `wds_port_dev` environment variable anymore

ryan-roemer commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,198 @@
+
+```console
+// This command bumps versions, runs checks, builds libs, and runs publishr postversion (~5 minutes)
+nps lerna-dry-run
````
Consistency nit: most shell commands start with `$ ` (same two lines down too).

ryan-roemer commented on config/webpack/demo/webpack.config.dev.js:
```diff
@@ -0,0 +1,62 @@
+        // Use include specifically of our sources.
+        // Do _not_ use an `exclude` here.
+        include: FILES.concat([DEMO]),
+        // **Note**: Cannot use shorthand `"babel-loader"` or `"babel"` when
```
NIT: You can remove this comment because modern webpack always requires full package name now for loaders.

ryan-roemer commented on config/webpack/webpack.config.test.js:
```diff
@@ -0,0 +1,43 @@
+  },
+  devtool: "source-map",
+  devServer: {
+    port: parseInt(process.env.npm_package_config_wds_port_test || WDS_PORT),
```
Same comment here.

ryan-roemer commented on demo/app.js:
```diff
@@ -1,6 +1,143 @@
+  }
+
+  getDemo() { // eslint-disable-line complexity
+    let Child;
```
NIT: I would have done an object map instead like:

```js
const MAP = {
  "/axis": AxisDemo,
  // ...
};

getDemo() {
  return MAP[this.state.route] || Home;
}
```

to avoid the monster `switch` statement.

ryan-roemer commented on .storybook/addons.js:
```diff
@@ -0,0 +1,2 @@
+import '@storybook/addon-options/register';
```
NIT: Are these linted? Seeing different quotation marks than the rest of the stuff.

ryan-roemer commented on package.json:
```diff
@@ -1,66 +1,111 @@
+    "chromatic": "chromatic test --storybook-addon",
+    "chromatic-ci": "chromatic test --storybook-addon --exit-zero-on-changes"
   },
   "devDependencies": {
```
Have you tried a git npm install like:

```sh
$ npm install FormidableLabs/victory#pre-1.0.0
```

?

When you do that you actually shouldn't get **any** devDependencies, so the `postinstall` should fail. Worth considering what our "git dependency" story is now that we're going monorepo, including dropping support all together.

/cc @tptee

boygirl commented:
@ryan-roemer I've addressed most of the comments you left with the following exception

- Not supporting git installs: I think think supporting them for `victory` and not supporting them for `victory-*` is confusing. I'd prefer to deprecate them all together

ryan-roemer commented on package.json:
```diff
@@ -1,66 +1,106 @@
     "url": "https://github.com/formidablelabs/victory/issues"
   },
-  "homepage": "https://github.com/formidablelabs/victory",
+  "homepage": "https://github.com/formidablelabs/open-source/victory",
```
I think this link is incorrect.

ryan-roemer commented on .gitignore:
```diff
@@ -35,5 +35,5 @@ lib
 node_modules
 npm-debug.log*
-yarn.lock
+lerna-debug.log*
```
Should also gitignore + git remove `package-lock.json` (It's also got a security warning flagged in GH web interface)

ryan-roemer commented on config/karma/karma.conf.js:
```diff
@@ -0,0 +1,57 @@
+      // Sinon has issues with webpack. Do global include.
+      require.resolve("sinon/pkg/sinon"),
+
+      // Polyfills for PhantomJS in React 16.
```
NIT: No longer needed for Phantom. _But_ if you test in IE9 you will need both. Can drop otherwise.

ryan-roemer commented on config/karma/karma.conf.dev.js:
```diff
@@ -0,0 +1,53 @@
+      // Sinon has issues with webpack. Do global include.
+      require.resolve("sinon/pkg/sinon"),
+
+      // Polyfills for PhantomJS in React 16.
```
NIT: No longer needed for Phantom. _But_ if you test in IE9 you will need both. Can drop otherwise.

boygirl commented:
@ryan-roemer yes, thank you for the review! I will plan on attempting a release as `30.0.0-alpha` just to be sure everything goes off without a hitch.
